packages/wallets/src/features/cashier/modules/TransferForm/components/TransferFormAccountSelection/TransferFormAccountSelection.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { useMemo } from 'react';
+import classNames from 'classnames';
+import { WalletText } from '../../../../../../components';
+import { useModal } from '../../../../../../components/ModalProvider';
+import useDevice from '../../../../../../hooks/useDevice';
+import CloseIcon from '../../../../../../public/images/close-icon.svg';
+import type { TAccount, TAccountList } from '../../types';
+import { TransferFormAccountCard } from '../TransferFormAccountCard';
+import './TransferFormAccountSelection.scss';
+
+type TProps = {
+    accountsList: TAccountList;
+    activeWallet: TAccount;
+    fromAccount?: TAccount;
+    label: 'Transfer from' | 'Transfer to';
+    onSelect: (value?: TAccount) => void;
+    selectedAccount?: TAccount;
+    toAccount?: TAccount;
+};
+
+const TitleLine = () => <div className='wallets-transfer-form-account-selection__title-line' />;
+
+const TransferFormAccountSelection: React.FC<TProps> = ({
+    accountsList,
+    activeWallet,
+    fromAccount,
+    label,
+    onSelect,
+    selectedAccount,
+    toAccount,
+}) => {
+    const { isMobile } = useDevice();
+    const modal = useModal();
+
+    const transferToHint = useMemo(() => {
+        const isTransferToHintVisible = label === 'Transfer to' && toAccount?.loginid === activeWallet?.loginid;
+
+        return isTransferToHintVisible
+            ? `You can only transfers funds from the ${fromAccount?.accountName} to the linked ${activeWallet?.accountName}.`
+            : '';
+    }, [activeWallet?.accountName, activeWallet?.loginid, fromAccount?.accountName, label, toAccount?.loginid]);
+
+    const isSingleAccountsGroup = useMemo(
+        () =>
+            (Object.keys(accountsList) as (keyof typeof accountsList)[]).filter(key => accountsList[key].length > 0)
+                .length === 1,
+        [accountsList]
+    );
+
+    return (
+        <div className='wallets-transfer-form-account-selection'>
+            <div className='wallets-transfer-form-account-selection__header'>
+                <WalletText size='md' weight='bold'>
+                    {label}
+                </WalletText>
+                <button className='wallets-transfer-form-account-selection__close-button' onClick={() => modal.hide()}>
+                    <CloseIcon />
+                </button>
+            </div>
+            <div className='wallets-transfer-form-account-selection__accounts'>
+                {(Object.keys(accountsList) as (keyof typeof accountsList)[]).map((key, index) => {
```

Comment:
`Object.entries` might be more concise

Reply:
yeah, @rostislav-deriv with `Object.entries` it looks much clearly ! 👍